pandas/core/internals/managers.py

```diff
@@ -106,7 +107,12 @@ class BlockManager(PandasObject):
 
     def __init__(self, blocks, axes, do_integrity_check=True):
         self.axes = [ensure_index(ax) for ax in axes]
-        self.blocks = tuple(blocks)
+
+        non_native_byteorder = '>' if sys.byteorder == 'little' else '<'
```

Comment:
this comparison should be in pandas/compat and done at import time

Reply:
Should I just add a function like get_non_native_byteorder() containing this line in the init of pandas/compat? Or should I rather use the is_platform_little_endian() function from compat here?